### PR TITLE
Redirection HTTP

### PR DESCRIPTION
**Pull request by ghost**
# Description
HTTP redirection are not accepted on OPTIONS by web browers.

# Features
- GET method is redirect with `302 Found` and location header
- OPTIONS method is handle and create a sub request to the target service to retrieve headers and send it back to the client
- Other methods are redirect with `307 Temporary Redirect` and a default CORS if `Origin` is asked

### CODE REVIEW

File: local_testing/README.MD [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Setup
+For Windows user, add this following line into the file `C:\Windows\System32\drivers\etc\hosts`
+
+```
+127.0.0.1 api.massive-dynamic.com
+127.0.0.1 api.ja.massive-dynamic.com
+127.0.0.1 api.na.massive-dynamic.com
+```
+
+For Linux user, add this following line into /etc/hosts
```
Comment (tmeunier) [inline local_testing/README.MD]: `Unix`? ;)

File: middleware.go [hunk 2]
```diff
@@ -126,6 +139,12 @@ func New(_ context.Context, next http.Handler, config *Config, name string) (htt
 		Log.LogDebug("Endianness=big")
 	}
 
+	routingMethod := RoutingMethodRedirect
+	if config.RoutingMethod == RoutingMethodDirectCall {
+		routingMethod = RoutingMethodDirectCall
+	}
+	Log.LogDebug(fmt.Sprintf("RoutingMethod configure to %s", routingMethod))
```
Comment (tmeunier) [inline middleware.go]: `configured`

File: middleware.go [hunk 3]
```diff
@@ -245,38 +270,99 @@ func handleJwtRedirection(req *http.Request,
 	return nil, nil
 }
 
+func sendHTTPRequest(rw http.ResponseWriter, req *http.Request, destination string) {
```
Comment (tmeunier) [inline middleware.go]: Wouldn't it be clearer if you named the function `proxyHTTPRequest`?

**Review by tmeunier — APPROVED** [review]
Looks good to me, minor changes could improve readability

File: local_testing/README.MD (hunk 1 above)
Comment (ghost) [reply local_testing/README.MD]: I don't know on MacOs or BSD, but we can assume this is the same

**Pull request merged**